src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -586,6 +586,22 @@ protected DynamoModel(IStartConfiguration config)
             Scheduler = new DynamoScheduler(thread, config.ProcessMode);
             Scheduler.TaskStateChanged += OnAsyncTaskStateChanged;
 
+            //TODO dispose this.
+            List < Action > requestedDisposalCalls = new List<Action>();
+            ExecutionEvents.RequestRunOnDynamoScheduler += (actionToSchedule) =>
+            {
+                requestedDisposalCalls.Add(actionToSchedule);
```

Comment:
This language assumes that every client that uses the RequestRunOnDynamoScheduler event to schedule an action will be doing so to request object disposal. If we're going to make that assumption, the naming should be consistent at all levels. If we're instead providing a general mechanism (which seems like a better idea, even if we restrict it to this use case initially) then that language should be consistent throughout as well.

Reply:
@Dewb totally agree.
I think the general mechanism and naming are where this should end up.